contracts/HumaniqToken.sol

```diff
@@ -63,76 +70,80 @@ contract HumaniqToken is StandardToken, SafeMath {
     function issueTokens(address _for, uint tokenCount)
         external
         payable
-        isCrowdfundingContract
+        onlyMinter
         returns (bool)
     {
         if (tokenCount == 0) {
             return false;
         }
-        
-        if (icoFinished && add(totalSupply, tokenCount) > maxTotalSupply) {
-          throw;
+
+        if (add(totalSupply, tokenCount) > maxTotalSupply) {
+            throw;
         }
-        
+
         totalSupply = add(totalSupply, tokenCount);
         balances[_for] = add(balances[_for], tokenCount);
         Issuance(_for, tokenCount);
         return true;
     }
 
-    function transfer(address _to, uint256 _value)
-        unlocked
-        returns (bool success)
-    {
-        return super.transfer(_to, _value);
-    }
-
-    function transferFrom(address _from, address _to, uint256 _value)
-        unlocked
-        returns (bool success)
-    {
-        return super.transferFrom(_from, _to, _value);
-    }
-
     /// @dev Function to change address that is allowed to do emission.
     /// @param newAddress Address of new emission contract.
-    function changeEmissionContractAddress(address newAddress)
-        external
+    function changeMinter(address newAddress)
+        public
         onlyFounder
         returns (bool)
     {
-        emissionContractAddress = newAddress;
+        minter = newAddress;
+
+        // Allow emission contract to distribute tokens minted during ICO stage
+        allowed[allocationAddressICO][minter] = ICOSupply;
     }
 
-    /// @dev Function that locks/unlocks transfers of token.
-    /// @param value True/False
-    function lock(bool value)
-        external
+    /// @dev Function to change founder address.
+    /// @param newAddress Address of new founder.
+    function changeFounder(address newAddress)
+        public
         onlyFounder
+        returns (bool)
     {
-        locked = value;
+        founder = newAddress;
+
+        // Allow founder to distribute tokens minted during preICO stage
+        allowed[allocationAddressPreICO][founder] = preICOSupply;
```

Comment:
It would be definitely more consistent, but we have pre-ICO tokens already distributed on previous token. Therefore we just need to copy balances from previous token to this one, without applying any logic to transactions like bonuses etc. I thought about creating another special contract for this, but decided that it would only complicated the overall structure.